code/modules/antagonists/traitor/contractor/datums/syndicate_contract.dm

```diff
@@ -0,0 +1,518 @@
+	if(!U || !M || status != CONTRACT_STATUS_ACTIVE)
+		invalidate()
+		return
+	else if(!F)
+		U.message_holder("Extraction flare could not be located, agent. Ensure the extraction zone is clear before signaling us.", 'sound/machines/terminal_prompt_deny.ogg')
+		return
+	else if(!ismob(contract.target?.current))
+		invalidate()
+		return
+	U.message_holder("Extraction signal received, agent. [GLOB.using_map.full_name]'s redspace transport jamming systems have been sabotaged. "\
+			 	   + "We have opened a temporary portal at your flare location - proceed to the target's extraction by inserting them into the portal.", 'sound/effects/confirmdropoff.ogg')
+	// Open a portal
+	var/obj/effect/portal/redspace/contractor/P = new(get_turf(F), pick(GLOB.syndieprisonwarp), null, 0)
+	P.contract = src
+	P.contractor_mind = M.mind
+	P.target_mind = contract.target
+	extraction_portal = P
+	do_sparks(4, FALSE, P.loc)
+
+/**
+  * Called when a contract target has been extracted through the portal.
+  *
+  * Arguments:
+  * * M - The target mob.
+  * * P - The extraction portal.
+  */
+/datum/syndicate_contract/proc/target_received(mob/M, obj/effect/portal/redspace/contractor/P)
+	complete(M.stat == DEAD)
+	handle_target_experience(M, P)
+	clean_up()
+
+/**
+  * Notifies the uplink's holder that a contract has been completed.
+  *
+  * Arguments:
+  * * tc - How many telecrystals they have received.
+  * * creds - How many credits they have received.
+  * * target_dead - Whether the target was extracted dead.
+  */
+/datum/syndicate_contract/proc/notify_completion(tc = 0, creds = 0, target_dead = FALSE)
+	var/penalty_text = ""
+	if(target_dead)
+		penalty_text = " (penalty applied as the target was extracted dead)"
+	owning_hub.uplink?.message_holder("Well done, agent. The package has been received and will be processed shortly before being returned. "\
+									 + "As agreed, you have been credited with [tc] telecrystals[penalty_text] and [creds] credits.", 'sound/machines/terminal_prompt_confirm.ogg')
+
+/**
+  * Handles the target's experience from extraction.
+  *
+  * Arguments:
+  * * M - The target mob.
+  * * P - The extraction portal.
+  */
+/datum/syndicate_contract/proc/handle_target_experience(mob/M, obj/effect/portal/redspace/contractor/P)
+	var/turf/T = get_turf(P)
+	var/mob/living/carbon/human/H = M
+
+	// Prepare their return
+	addtimer(CALLBACK(src, .proc/handle_target_return, M, T), prison_time)
+
+	// Shove all of the victim's items in the secure locker.
+	victim_belongings = list()
+	var/list/obj/item/stuff_to_transfer = list()
+	for(var/obj/item/I in M)
+		// Any items we don't want to take from them?
+		if(istype(H))
+			// Keep their uniform and shoes
+			if(I == H.w_uniform || I == H.shoes)
+				continue
+			// Plasmamen are no use if they're crispy
+			if(isplasmaman(H) && I == H.head)
+				continue
+
+		// Any kind of implant gets removed (mindshield, etc)
+		if(istype(I, /obj/item/implant))
+			qdel(I)
```

Comment:
what happens to, say, extended mop implant (that gives you a NODROP mop)? Is it stuck to your hand? (i see you're not dropping NODROP items a bit lower)
Same question about the "normal" nodrop implant

Reply:
Yep those will break. Though I recon that that needs to be handled on the implant's `Destroy`

Reply:
These are not cybernetic implants (as they are handled below) however, leg cyber implants retract any extended item upon removal. Anti-drop implants are supposed to make hands relax on removal, but failed due to a runtime, which is now fixed.